frontend/webadmin/modules/uicommonweb/src/main/java/org/ovirt/engine/ui/uicommonweb/models/storage/UploadImageModel.java

```diff
@@ -161,30 +160,27 @@ public void setTestResponse(EntityModel<Response> testResponse) {
     public UploadImageModel(final Guid limitToStorageDomainId, final DiskImage resumeUploadDisk) {
         if (resumeUploadDisk == null) {
             setDiskModel(new NewDiskModel() {
+                private final boolean isChangeable = limitToStorageDomainId == null;
+
                 @Override
                 public void initialize() {
                     super.initialize();
 
-                    getStorageDomain().setIsChangeable(limitToStorageDomainId == null);
-                    getDataCenter().setIsChangeable(limitToStorageDomainId == null);
+                    getStorageDomain().setIsChangeable(isChangeable);
+                    getDataCenter().setIsChangeable(isChangeable);
                     getStorageType().setIsChangeable(false);
                     getSize().setIsChangeable(false);
                 }
 
                 @Override
                 protected void updateStorageDomains(final StoragePool datacenter) {
-                    if (limitToStorageDomainId == null) {
+                    if (isChangeable) {
                         super.updateStorageDomains(datacenter);
                     } else {
-                        AsyncDataProvider backend = AsyncDataProvider.getInstance();
-                        backend.getStorageDomainById(
+                        AsyncDataProvider.getInstance().getStorageDomainById(
                                 new AsyncQuery<>(storageDomain -> {
                                     getStorageDomain().setSelectedItem(storageDomain);
```

Comment:
`getStorageDomain().setSelectedItem()` at this point happens after the main initialization  - we have a chain of async actions (no race) which is OK. However as mentioned above SD list seems still not initialized. The other issue is that the selected DC is not connected with the selected SD. The [selection](https://github.com/oVirt/ovirt-engine/blob/f3cf526a05d5b414800ba3e1fd7bcb3438eeac91/frontend/webadmin/modules/uicommonweb/src/main/java/org/ovirt/engine/ui/uicommonweb/models/vms/AbstractDiskModel.java#L462) is dependent from filtering logic (`dataCenter.getStatus() == StoragePoolStatus.Up`) and from the disk image (`getDiskImage().getStoragePoolId())`.